extensions/desktop/common/init

```diff
@@ -6,6 +6,10 @@
 # shellcheck disable=SC2034
 START=$(date +%s.%N)
 
+function ensure_dir_exists() {
+  [ -d "$1" ] ||  mkdir -p "$@"
```

Comment:
Very simple, nice!

Beware that `$1` vs `$@` behavior is different.

I would suggest either allowing it to take multiple args (to match mkdir), or require just one.  A quick grep looks like all of the current use cases are single directory.

Reply:
I use `$@` so that this function can pass additional flags to mkdir. This is needed for the following line.

https://github.com/snapcore/snapcraft/blob/45e6b956cf8f9904f7994bb5492c4aec97a82604/extensions/desktop/common/desktop-exports#L136

This means that the first argument to `ensure_dir_exists` must always be the path, and the following arguments are not allowed to be paths. Is it ok to just document this in a comment above the function?